src/dbnode/storage/index.go

```diff
@@ -1130,10 +1130,19 @@ func (i *nsIndex) flushBlockSegment(
 			}
 
 			for _, result := range results.Results() {
```

Comment:
We make a call to fetch series metadata here:
```
var (
        res             = s.opts.FetchBlocksMetadataResultsPool().Get()
        fetchCtx        = s.contextPool.Get()
        nextIndexCursor *int64
)

var loopErr error
s.forEachShardEntry(func(entry *lookup.Entry) bool {
        // Break out of the iteration loop once we've accumulated enough entries.
        if int64(len(res.Results())) >= limit {
                next := int64(entry.Index)
                nextIndexCursor = &next
                return false
        }

        // Fast forward past indexes lower than page token
        if int64(entry.Index) < indexCursor {
                return true
        }

        // Use a context here that we finalize immediately so the stream
        // readers can be returned to pool after we finish fetching the
        // metadata for this series.
        // NB(r): Use a pooled context for pooled finalizers/closers but
        // reuse so don't need to put and get from the pool each iteration.
        fetchCtx.Reset()
        metadata, err := entry.Series.FetchBlocksMetadata(ctx, start, end, opts)
```

It looks like the only usage is in the shard.
```
src/dbnode//storage/shard.go:1650:              metadata, err := entry.Series.FetchBlocksMetadata(ctx, start, end, opts)
```

Since the series itself has the `metadata` as a `doc.Document` already, it feels like we can just return a ref to that in the results. Although its already ref'ing the doc.document in a way like so:
```
tagsIter := s.opts.IdentifierPool().TagsIterator()
tagsIter.ResetFields(s.metadata.Fields)
return block.NewFetchBlocksMetadataResult(s.id, tagsIter, res), nil
```
We could check downstream usage and possibly defer creating a tags iterator, maybe like a method on the fetch results so that we don't have to add an extra call to the shard to do double work. 

Although IMO the memory savings is prob similar since we are doing some pooling on top of taking refs to the `doc.Document` metadata in the series.

Reply:
I lied, I forgot that we don't even touch that code path above since we fetch series data from disk during a warm flush. 

Let's test as is.